WordPress/gutenberg
# [RNMobile] Fix pasting HTML into the post title

## What?


Updates paste handler logic of the post title to prevent an exception/crash when pasting HTML content.

## Why?


This PR is needed to prevent an exception/crash.

## How?


If the first element of the HTML pasted in the post title is a paragraph or heading tag, the paste handler will obtain its content:
https://github.com/WordPress/gutenberg/blob/d791ac8ab3d9364d81d9e4c0b217e60a9d2d190a/packages/editor/src/components/post-title/index.native.js#L79

Similar to the issue addressed in https://github.com/WordPress/gutenberg/pull/57028, the value provided by the `RichText` can't be treated as a string. Hence, the above line will produce the exception/crash.

This has been solved by utilizing the `toString` function to get the content, this is the same approach used in https://github.com/WordPress/gutenberg/pull/57028.

Additionally, we introduced integration tests to cover this case and others related to pasting content in the post title. In relation to this, some modifications have been applied to the test helpers derived from the test implementation.

## Testing Instructions





1. Open the app.
1. Copy the following HTML from your device:
```
<h2>Howdy</h2>
<h2>This is a heading.</h2>
<p>This is a paragraph.</p>
```
3. Create a post.
4. Paste the the HTML.
5. Observe that no exceptions/crashes are produced.
6. Observe that the post title is populated with the content of the first heading (i.e.g `Howdy`).
7. Observe that the rest of the HTML content is rendered as blocks in the post content.

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 
N/A
